src/index.ts

```diff
@@ -0,0 +1,518 @@
+   * @param locale  The locale that should be used in the place of the alias.
+   */
+  public setAlias(alias: string | string[], locale: string): void {
+    // Normalize locales.
+    const aliases = this.normalizeLocales(alias);
+    const normalizedLocale = this.normalizeLocale(locale);
+
+    // Prevent recursive behavior.
+    if (normalizedLocale in this.aliases || aliases.includes(normalizedLocale)) {
+      throw new Error(`${locale} cannot already be an alias.`);
+    }
+
+    // Make sure no aliases have a translation.
+    aliases
+      .forEach((a) => {
+        if (this.hasTranslationsForLocale(a)) {
+          throw new Error(`${a} has translations set and should not be used as an alias.`);
+        }
+      });
+
+    const hasChanges =
+      aliases
+        .map((a) => {
+          if (this.aliases[a] !== normalizedLocale) {
+            this.aliases[a] = normalizedLocale;
+            return true;
+          } else {
+            return false;
+          }
+        })
+        .includes(true);
+
+    if (hasChanges) {
+      this.callSubscribers();
+    }
+  }
+
+  /**
+   * Sets a locale to fallback to when a string is unavailable for the main locale. Ideally, the
+   * fallback locale has a translation for every key.
+   *
+   * @param fallback  Locale that should be used when primary locale can't be found.
+   */
+  public setFallback(fallback: string): void {
+    const normalized = this.normalizeLocale(fallback);
+
+    if (this.fallback !== normalized) {
+      this.fallback = normalized;
+      this.callSubscribers();
+    }
+  }
+
+  /**
+   * Having key mode on will cause the `t` method to always return the key. This may be useful for
+   * debugging.
+   */
+  public setKeyMode(toggle: boolean): void {
+    if (this.keyMode !== toggle) {
+      this.keyMode = toggle;
+      this.callSubscribers();
+    }
+  }
+
+  /**
+   * Sets the locales to use when calling t(). By default, the method tries to find an ideal
+   * fallback sequence for the provided locale. For example, `['zh-Hant-HK', 'zh-Hant-TW']`
+   * would try to match: `['zh-Hant-HK', 'zh-Hant-TW', 'zh-Hant', 'zh']`.
+   *
+   * If passed an array of locales in order of preferences, the first locale that has translations
+   * will be chosen. If no translations are found, then locale will be unset and fallback will be
+   * used.
+   *
+   * This method should be called after setting up all rules and translations. If called before, a
+   * locale might be considered unavailable and will not be chosen.
+   */
+  public setLocale(locale: string | string[]): void {
+    // Normalize all the locales.
+    const locales: string[] = this.normalizeLocales(locale);
+
+    // If there is no match found while iterating through the list, use this one.
+    const possibleLocales: string[] = [];
+
+    // Create a new list of locales that will increase the odds of matching.
+    const baseSet: Set<string> = new Set(locales);
```

Comment:
According to https://babeljs.io/learn-es2015/#ecmascript-2015-features-map-set-weak-map-weak-set

> In order to support Maps, Sets, WeakMaps, and WeakSets in all environments you must include the Babel polyfill.

Maybe this should be mentioned in the README somewhere, or maybe this should be changed to use a plain object instead of an ES6 `Set`?

Reply:
Forgot that this isn't using babel, but I think the same still applies with TypeScript: it probably has type definitions for `Map`/`Set`/etc. built in, but doesn't polyfill them for you.